### ISSUE TITLE
extremely weird problem with 0.11.x - no save()-Callback

### ISSUE THREAD

**Issue by japel** [balderdashy/sails-mongo#324]
Hey there! I'm having really weird problems with 0.11.x version of sails-mongo:

The problem: save()-Callback is only sometimes called.

Tracking it down:

I tested this in bootstrap and as a dummy controller function called by a test route:
```javascript
Test.create({name: "test", age: 25})
      .exec(function (error, test) {
        Test1.create({name: "test1", age: 23})
          .exec(function(error,test1){
            test.test1.add(test1.id);
            test.save(function(error,saved){
              console.log(saved);
            });
          });
      });
```
Both work, saved is logged to console, no matter which version of sails-mongo I use, which is what I expected.

Then I tested the following with my UserController (and a route):
```javascript
addGroup: function (req, res) {
    var id = req.param('id');
    var groupId = req.param('groupId');
    if (!id || !groupId) {
      return res.badRequest('No id provided');
    }
    User.findOne(id).exec(function (err, found) {
      if (err) return res.serverError(err);
      if (!found) return res.notFound();
      found.groups.add(groupId);
      found.save(function (error, res_user) {
        if (error) return res.json(error);
        res.status(201);
        res.json(res_user);
      });
    });
  },
```
sails-mongo@0.11.2 - With the debugger I get to found.save(), but then my RAM-usage goes up to several gigabytes and the save()-callback is never called. This also happens with 0.11.1 and 0.11.0
sails-mongo@0.10.8 - works as expected, I get the user as response.

I couldn't believe my eyes. So I tested the above code with 0.11.2 but in bootstrap and with static IDs:

```javascript
  User.findOne("5466010516e6e9a445ad83c0").exec(function (err, found) {
    if (err) return res.serverError(err);
    if (!found) return res.notFound();
    else {
      found.groups.add("5592493add03aeac47b30a3c");
    }
    found.save(function (error, res_user) {
      if (error) return console.log(error);
      console.log(res_user);
    });
  });
```
And it works, no matter which version of sails-mongo I use.
So it seems to have something to do with sails-mongo version combined with controllers?!

But wait! It gets weirder:
If I lift sails with a new empty db, I am able to save() no matter what version of sails-mongo I use.

Does anyone ever had a similar problem?

**Comment by japel** [comment]
Now I took a look at the mongodb logs when calling my addGroup method with 0.10.8 and this is what I get:
```
Tue Aug 25 12:44:44.167 [conn129] query myDatabase.user query: { _id: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1 idhack:1 keyUpdates:0 locks(micros) r:164 reslen:311 0ms
Tue Aug 25 12:44:44.171 [conn130] query myDatabase.user query: { _id: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:0 idhack:1 keyUpdates:0 locks(micros) r:57 reslen:311 0ms
Tue Aug 25 12:44:44.174 [conn134] update myDatabase.user query: { _id: ObjectId('5466010xxxe6e9a445adxxxx') } update: { $set: { GUID: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX", active: true, createdAt: new Date(1415971077751), email: "xxx.xxx@xxx.xx", firstTime: false, firstname: "XXX", from: "xxx", lastname: "xxx", xxx: false, sessionId: "XXXXX", updatedAt: new Date(1440499484166), username: "xxx" } } nscanned:1 nupdated:1 keyUpdates:0 locks(micros) w:367 0ms
Tue Aug 25 12:44:44.174 [conn134] run command myDatabase.$cmd { getlasterror: 1 }
Tue Aug 25 12:44:44.174 [conn134] command myDatabase.$cmd command: { getlasterror: 1 } ntoreturn:1 keyUpdates:0  reslen:85 0ms
Tue Aug 25 12:44:44.175 [conn135] query myDatabase.user query: { _id: { $in: [ ObjectId('5466010xxxe6e9a445adxxxx') ] } } ntoreturn:0 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:267 nreturned:1 reslen:311 0ms
Tue Aug 25 12:44:44.178 [conn130] insert myDatabase.history ninserted:1 keyUpdates:0 locks(micros) w:355 0ms
Tue Aug 25 12:44:44.178 [conn130] run command myDatabase.$cmd { getlasterror: 1 }
Tue Aug 25 12:44:44.178 [conn130] command myDatabase.$cmd command: { getlasterror: 1 } ntoreturn:1 keyUpdates:0  reslen:67 0ms
Tue Aug 25 12:44:44.181 [conn133] query myDatabase.group query: { _id: ObjectId('5592493addxxxaeacxxxx0a3c') } ntoreturn:1 idhack:1 keyUpdates:0 locks(micros) r:122 reslen:159 0ms
Tue Aug 25 12:44:44.182 [conn134]   running multiple plans
Tue Aug 25 12:44:44.182 [conn134] query myDatabase.group_members__user_groups query: { group_members: ObjectId('5592493addxxxaeacxxxx0a3c'), user_groups: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1 ntoskip:0 nscanned:2 keyUpdates:0 locks(micros) r:495 nreturned:0 reslen:20 0ms
Tue Aug 25 12:44:44.208 [conn129] insert myDatabase.group_members__user_groups ninserted:1 keyUpdates:0 locks(micros) w:20461 23ms
Tue Aug 25 12:44:44.208 [conn129] run command myDatabase.$cmd { getlasterror: 1 }
Tue Aug 25 12:44:44.209 [conn129] command myDatabase.$cmd command: { getlasterror: 1 } ntoreturn:1 keyUpdates:0  reslen:67 0ms
Tue Aug 25 12:44:44.213 [conn130] query myDatabase.user query: { _id: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:0 idhack:1 keyUpdates:0 locks(micros) r:57 reslen:311 0ms
Tue Aug 25 12:44:44.215 [conn133] query myDatabase.passport query: { user: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:0 ntoskip:0 nscanned:2 keyUpdates:0 locks(micros) r:242 nreturned:0 reslen:20 0ms
Tue Aug 25 12:44:44.216 [conn134] query myDatabase.right query: { user: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:0 ntoskip:0 nscanned:186 keyUpdates:0 locks(micros) r:516 nreturned:0 reslen:20 0ms
Tue Aug 25 12:44:44.216 [conn135] query myDatabase.group_members__user_groups query: { user_groups: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:0 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:176 nreturned:1 reslen:94 0ms
Tue Aug 25 12:44:44.217 [conn130] query myDatabase.option query: { _id: null } ntoreturn:0 ntoskip:0 nscanned:0 keyUpdates:0 locks(micros) r:135 nreturned:0 reslen:20 0ms
Tue Aug 25 12:44:44.217 [conn129] query myDatabase.generic query: { _id: null } ntoreturn:0 ntoskip:0 nscanned:0 keyUpdates:0 locks(micros) r:313 nreturned:0 reslen:20 0ms
Tue Aug 25 12:44:44.218 [conn133] query myDatabase.group query: { _id: { $in: [ ObjectId('5592493addxxxaeacxxxx0a3c') ] } } ntoreturn:0 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:251 nreturned:1 reslen:159 0ms
```

And this is what I get with 0.11.2:
```
Tue Aug 25 12:52:35.939 [conn160] query myDatabase.user query: { _id: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1000 idhack:1 keyUpdates:0 locks(micros) r:50 reslen:311 0ms
Tue Aug 25 12:52:35.941 [conn161] query myDatabase.passport query: { user: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1000 ntoskip:0 nscanned:2 keyUpdates:0 locks(micros) r:237 nreturned:0 reslen:20 0ms
Tue Aug 25 12:52:35.942 [conn157] query myDatabase.right query: { user: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1000 ntoskip:0 nscanned:186 keyUpdates:0 locks(micros) r:446 nreturned:0 reslen:20 0ms
Tue Aug 25 12:52:35.942 [conn160] query myDatabase.option ntoreturn:1000 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:47 nreturned:1 reslen:105 0ms
Tue Aug 25 12:52:35.944 [conn158] query myDatabase.group_members__user_groups query: { user_groups: ObjectId('5466010xxxe6e9a445adxxxx') } ntoreturn:1000 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:212 nreturned:1 reslen:94 0ms
Tue Aug 25 12:52:35.945 [conn159] query myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 ntoskip:0 nscanned:1001 keyUpdates:0 locks(micros) r:1928 nreturned:1000 reslen:493644 1ms
Tue Aug 25 12:52:35.945 [conn161] query myDatabase.group query: { _id: { $in: [ ObjectId('5592493addxxxaeacxxxx0a3c') ] } } ntoreturn:1000 ntoskip:0 nscanned:1 keyUpdates:0 locks(micros) r:209 nreturned:1 reslen:159 0ms
Tue Aug 25 12:52:36.024 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1310 nreturned:1000 reslen:399316 1ms
Tue Aug 25 12:52:36.095 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1218 nreturned:1000 reslen:301880 1ms
Tue Aug 25 12:52:36.148 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1299 nreturned:1000 reslen:302104 1ms
Tue Aug 25 12:52:36.207 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1235 nreturned:1000 reslen:302911 1ms
Tue Aug 25 12:52:36.259 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1225 nreturned:1000 reslen:301874 1ms
Tue Aug 25 12:52:36.309 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1185 nreturned:1000 reslen:302477 1ms
Tue Aug 25 12:52:36.370 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1271 nreturned:1000 reslen:303835 1ms
Tue Aug 25 12:52:36.421 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1223 nreturned:1000 reslen:305940 1ms
Tue Aug 25 12:52:36.472 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1321 nreturned:1000 reslen:304574 1ms
Tue Aug 25 12:52:36.523 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1132 nreturned:1000 reslen:304452 1ms
Tue Aug 25 12:52:36.581 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1094 nreturned:1000 reslen:304462 1ms
Tue Aug 25 12:52:36.632 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1152 nreturned:1000 reslen:305246 1ms
Tue Aug 25 12:52:36.682 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1164 nreturned:1000 reslen:304660 1ms
Tue Aug 25 12:52:36.699 [conn98] run command admin.$cmd { ping: 1 }
Tue Aug 25 12:52:36.700 [conn98] command admin.$cmd command: { ping: 1 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:36.702 [conn97] run command myDatabase.$cmd { ping: 1.0 }
Tue Aug 25 12:52:36.702 [conn97] command myDatabase.$cmd command: { ping: 1.0 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:36.734 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1107 nreturned:1000 reslen:304798 1ms
Tue Aug 25 12:52:36.801 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:5518 nreturned:1000 reslen:305138 5ms
Tue Aug 25 12:52:36.856 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1247 nreturned:1000 reslen:404565 1ms
Tue Aug 25 12:52:36.902 [conn56] run command admin.$cmd { ping: 1 }
Tue Aug 25 12:52:36.903 [conn56] command admin.$cmd command: { ping: 1 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:36.904 [conn65] run command myDatabase.$cmd { ping: 1.0 }
Tue Aug 25 12:52:36.904 [conn65] command myDatabase.$cmd command: { ping: 1.0 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:36.938 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1306 nreturned:1000 reslen:405464 1ms
Tue Aug 25 12:52:37.007 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1265 nreturned:1000 reslen:412937 1ms
Tue Aug 25 12:52:37.091 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1239 nreturned:1000 reslen:403583 1ms
Tue Aug 25 12:52:37.160 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1273 nreturned:1000 reslen:405931 1ms
Tue Aug 25 12:52:37.249 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1427 nreturned:1000 reslen:408384 1ms
Tue Aug 25 12:52:37.319 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1255 nreturned:1000 reslen:415020 1ms
Tue Aug 25 12:52:37.406 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:5471 nreturned:1000 reslen:412919 5ms
Tue Aug 25 12:52:37.481 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1376 nreturned:1000 reslen:421779 1ms
Tue Aug 25 12:52:37.559 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1231 nreturned:1000 reslen:419855 1ms
Tue Aug 25 12:52:37.634 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1195 nreturned:1000 reslen:415026 1ms
Tue Aug 25 12:52:37.711 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1192 nreturned:1000 reslen:424426 1ms
Tue Aug 25 12:52:37.785 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1198 nreturned:1000 reslen:422103 1ms
Tue Aug 25 12:52:37.886 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1227 nreturned:1000 reslen:418578 1ms
Tue Aug 25 12:52:37.964 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1327 nreturned:1000 reslen:421968 1ms
Tue Aug 25 12:52:38.156 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1304 nreturned:1000 reslen:429149 1ms
Tue Aug 25 12:52:38.231 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1222 nreturned:1000 reslen:423333 1ms
Tue Aug 25 12:52:38.305 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1388 nreturned:1000 reslen:423471 1ms
Tue Aug 25 12:52:38.390 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1226 nreturned:1000 reslen:404145 1ms
Tue Aug 25 12:52:38.458 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1192 nreturned:1000 reslen:402536 1ms
Tue Aug 25 12:52:38.537 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1191 nreturned:1000 reslen:401905 1ms
Tue Aug 25 12:52:38.621 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1186 nreturned:1000 reslen:400574 1ms
Tue Aug 25 12:52:38.665 [conn67] run command admin.$cmd { ping: 1 }
Tue Aug 25 12:52:38.666 [conn67] command admin.$cmd command: { ping: 1 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:38.667 [conn66] run command myDatabase.$cmd { ping: 1.0 }
Tue Aug 25 12:52:38.667 [conn66] command myDatabase.$cmd command: { ping: 1.0 } ntoreturn:1 keyUpdates:0  reslen:37 0ms
Tue Aug 25 12:52:38.690 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1354 nreturned:1000 reslen:405859 1ms
Tue Aug 25 12:52:38.772 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1833 nreturned:1000 reslen:417918 1ms
Tue Aug 25 12:52:38.844 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1383 nreturned:1000 reslen:410948 1ms
Tue Aug 25 12:52:38.925 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1344 nreturned:1000 reslen:413515 1ms
Tue Aug 25 12:52:38.999 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1240 nreturned:1000 reslen:408173 1ms
Tue Aug 25 12:52:39.084 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1264 nreturned:1000 reslen:407647 1ms
Tue Aug 25 12:52:39.163 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1231 nreturned:1000 reslen:417247 1ms
Tue Aug 25 12:52:39.242 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1193 nreturned:1000 reslen:405688 1ms
Tue Aug 25 12:52:39.312 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1298 nreturned:1000 reslen:407549 1ms
Tue Aug 25 12:52:39.391 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1229 nreturned:1000 reslen:408898 1ms
Tue Aug 25 12:52:39.459 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1205 nreturned:1000 reslen:414369 1ms
Tue Aug 25 12:52:39.528 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1329 nreturned:1000 reslen:415142 1ms
Tue Aug 25 12:52:39.611 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1195 nreturned:1000 reslen:406295 1ms
Tue Aug 25 12:52:39.681 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1212 nreturned:1000 reslen:414933 1ms
Tue Aug 25 12:52:39.764 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1591 nreturned:1000 reslen:805309 1ms
Tue Aug 25 12:52:39.899 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1728 nreturned:1000 reslen:1016244 1ms
Tue Aug 25 12:52:40.069 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:2702 nreturned:1000 reslen:1429240 2ms
Tue Aug 25 12:52:40.311 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1766 nreturned:1000 reslen:957562 1ms
Tue Aug 25 12:52:40.615 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1463 nreturned:1000 reslen:638319 1ms
Tue Aug 25 12:52:40.714 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1488 nreturned:1000 reslen:714485 1ms
```

This line:
```
Tue Aug 25 12:52:40.714 [conn159] getmore myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 keyUpdates:0 locks(micros) r:1488 nreturned:1000 reslen:714485 1ms
```
repeats for nearly forever and causes my RAM to explode.

What is happening there?

**Comment by devinivy** [comment]
Thanks– you've done some nice snooping.  Seems like there's a genuine bug here.  Do you think this may be the same problem as in #300?

**Comment by japel** [comment]
@devinivy I'm unfortunately not that deep into waterline/sails-mongo to tell so, but it seems so.
in #300 @diegoaguilar is doing an update(), which internally probably calls save() somewhere (or vice versa).

**Comment by devinivy** [comment]
Can you check your logs for a query preceding the `getmore` ops that would try to return all records from the db?  I wonder if criteria is messed-up somewhere, causing a giant `find`.

**Comment by japel** [comment]
yeah it looks like a giant find(), that explains my RAM-exploding
there is no other presiding query querying the collection generic besides the one using that cursor.
```
Tue Aug 25 12:52:35.945 [conn159] query myDatabase.generic cursorid:10921762685240111 ntoreturn:1000 ntoskip:0 nscanned:1001 keyUpdates:0 locks(micros) r:1928 nreturned:1000 reslen:493644 1ms
```

I also just wonder why it is querying that collection: neither user nor group has anything to do with generic...

**Comment by devinivy** [comment]
@japel you indicated in #300 that there may be a driver issue.  Does it make sense that a driver bug is the cause of this issue as well?

**Comment by japel** [comment]
@devinivy I'm not sure if it's the same Issue, but we were able to get rid of it when manually changing mongodb driver version

**Comment by devinivy** [comment]
Thanks!  @tjwebb any suggestions how to best address this?  File a bug with the driver?  Version the driver more explicitly in deps?  Note in the docs?

**Comment by japel** [comment]
I've tested the above with mongodb 2.0.42 and the symptoms seem to go away

**Comment by devinivy** [comment]
That's great news!

**Comment by devinivy** [comment]
Should be fixed with #332.
